src/endpoints/authentication/impl.cpp

```diff
@@ -20,120 +21,484 @@
+				                                       url = std::move(url)] {
+					// Two query params requiured by OAuth 2.0
+					// TODO: Double check with OIDC for bonus params
+					const auto code_iter{url.query.find("code")};
+					const auto state_iter{url.query.find("state")};
+
+					// Check to see if querys are valid
+					if (state_iter == std::end(url.query) || code_iter == std::end(url.query)) {
+						return _sess_ptr->send(fail(status_type::bad_request));
+					}
+
+					// Code here...
+					// Verify the state here!!!
+					log::debug("{}: Code is [{}]", fn, code_iter->second);
+					log::debug("{}: State is [{}]", fn, state_iter->second);
+
+					// Populate arguments
+					BodyArguments args{
+						{"grant_type", "authorization_code"},
+						{"client_id",
+					     irods::http::globals::oidc_configuration().at("client_id").get_ref<const std::string&>()},
+						{"code", code_iter->second},
+						{"redirect_uri",
+					     irods::http::globals::oidc_configuration().at("redirect_uri").get_ref<const std::string&>()}};
+
+					// Encode the string, hit endpoint, get res
+					nlohmann::json res_item{hit_token_endpoint(encode_body(args))};
+
+					// Assume passed, get oidc token
+					const std::string jwt_token{res_item.at("id_token").get_ref<const std::string&>()};
+
+					// Get OIDC token && feed to JWT parser
+					// TODO: Handle case where we throw!!!
+					auto decoded_token{
+						jwt::decode<jwt::traits::nlohmann_json>(res_item.at("id_token").get_ref<const std::string&>())};
+
+					// Get irods username
+					const std::string irods_name{
+						decoded_token.get_payload_json().at("irods_username").get<const std::string>()};
+
+					// Issue token?
+					static const auto seconds =
+						irods::http::globals::configuration()
+							.at(nlohmann::json::json_pointer{"/http_server/authentication/basic/timeout_in_seconds"})
+							.get<int>();
+
+					auto bearer_token = irods::process_stash::insert(authenticated_client_info{
+						.auth_scheme = authorization_scheme::basic,
+						.username = std::move(irods_name),
+						.expires_at = std::chrono::steady_clock::now() + std::chrono::seconds{seconds}});
+
+					response_type res_rep{status_type::ok, _req.version()};
+					res_rep.set(field_type::server, BOOST_BEAST_VERSION_STRING);
+					res_rep.set(field_type::content_type, "text/plain");
+					res_rep.keep_alive(_req.keep_alive());
+					res_rep.body() = std::move(bearer_token);
+					res_rep.prepare_payload();
+
+					return _sess_ptr->send(std::move(res_rep));
+				});
+			}
+		}
+		// Handle posts
+		else if (_req.method() == boost::beast::http::verb::post) {
+			irods::http::globals::background_task([fn = __func__, _sess_ptr, _req = std::move(_req)] {
+				// Right, we're kinda being a proxy, so how about proxy-authorization?
+				const auto& hdrs{_req.base()};
+				const auto iter{hdrs.find("authorization")};
+
+				// Failed to find auth header, use OIDC auth_code flow instead
```

Comment:
Incorrect comment?

Reply:
Likely artifact of the original OIDC grant implementation, before we used the 'iRODS' auth scheme.

Will remove.